### PR TITLE
CC use depmod in rootfs builder for kernel modules with docker support

### PR DESCRIPTION
**Pull request by Alex-Carter01**
This adds depmod to the rootfs builder to get the dependencies for kernel modules, namely for the efi_secret module needed for sev.

In order to maintain compatibility with the local build docker environment, a few dependencies are needed.

### CODE REVIEW

**Review by fitzthum — APPROVED** [review]
Nice.

File: tools/osbuilder/rootfs-builder/rootfs.sh [hunk 1]
```diff
@@ -273,6 +273,8 @@ copy_kernel_modules()
 	info "Copy kernel modules from ${KERNEL_MODULES_DIR}"
 	mkdir -p "${dest_dir}"
 	cp -a "${KERNEL_MODULES_DIR}" "${dest_dir}/"
+	local KERNEL_VER=$(ls ${dest_dir})
+    depmod -b "${rootfs_dir}" ${KERNEL_VER}
```
Comment (fidencio) [inline tools/osbuilder/rootfs-builder/rootfs.sh:277]: Use tab here, following the rest of the file.

File: tools/osbuilder/rootfs-builder/ubuntu/config.sh [hunk 2]
```diff
@@ -5,11 +5,11 @@
 OS_NAME=ubuntu
 # This should be Ubuntu's code name, e.g. "focal" (Focal Fossa) for 20.04
 OS_VERSION=${OS_VERSION:-focal}
-PACKAGES="chrony iptables"
+PACKAGES="chrony iptables kmod"
 [ "$AGENT_INIT" = no ] && PACKAGES+=" init"
 [ "$KATA_BUILD_CC" = yes ] && PACKAGES+=" cryptsetup-bin e2fsprogs"
 [ "$SECCOMP" = yes ] && PACKAGES+=" libseccomp2"
-[ "$SKOPEO" = yes ] && PACKAGES+=" libgpgme11"
+[ "$SKOPEO" = yes ] && PACKAGES+=" libgpgme11 libdevmapper1.02.1"
```
Comment (amshinde) [inline tools/osbuilder/rootfs-builder/ubuntu/config.sh:12]: Why is libdevmapper being added? Seems like an unrelated change.

File: tools/osbuilder/rootfs-builder/rootfs.sh [hunk 3]
```diff
@@ -274,7 +274,7 @@ copy_kernel_modules()
 	mkdir -p "${dest_dir}"
 	cp -a "${KERNEL_MODULES_DIR}" "${dest_dir}/"
 	local KERNEL_VER=$(ls ${dest_dir})
-    depmod -b "${rootfs_dir}" ${KERNEL_VER}
+	depmod -b "${rootfs_dir}" ${KERNEL_VER}
```
Comment (amshinde) [inline tools/osbuilder/rootfs-builder/rootfs.sh:277]: Please fix this in the previous commit.

**Comment by Alex-Carter01** [comment]
changed commit of tab fix (oops, thank you!)

looking back to see what ibdevmapper was for

File: tools/osbuilder/rootfs-builder/rootfs.sh [hunk 4]
```diff
@@ -273,6 +273,8 @@ copy_kernel_modules()
 	info "Copy kernel modules from ${KERNEL_MODULES_DIR}"
 	mkdir -p "${dest_dir}"
 	cp -a "${KERNEL_MODULES_DIR}" "${dest_dir}/"
+	local KERNEL_VER=$(ls ${dest_dir})
```
Comment (wainersm) [inline tools/osbuilder/rootfs-builder/rootfs.sh:276]: Any chance of having modules for more than one kernel on $dest_dir? If yes, than you should probably pick one or depmod can handle more than one version at once?

Comment (Alex-Carter01) [reply tools/osbuilder/rootfs-builder/rootfs.sh:276]: If this directory is the output of the build_image script, then there will be only one. But this could be used more generically if it is ran outside of the build_image script and it would be possible to have multiple directories, which would make a loop necesary.

File: tools/osbuilder/rootfs-builder/rootfs.sh (hunk 3 above)
Comment (Alex-Carter01) [reply tools/osbuilder/rootfs-builder/rootfs.sh:277]: moved to the first commit

**Comment by Alex-Carter01** [comment]
Changing commit message to help commit checker.

**Comment by wainersm** [comment]
Hi @Alex-Carter01 , need to fix a problem on the commit message: " Subject too long (max 75)"

**Comment by Alex-Carter01** [comment]
Reduced subject length and changed subsystem to osbuilder

**Comment by wainersm** [comment]
Hi @Alex-Carter01 , another conflict to fix before we land this one.

**Comment by Alex-Carter01** [comment]
Rebased to resolve conflict - came from addition of a dockerfile dependency from the agent config PR.

**Comment by wainersm** [comment]
@Alex-Carter01 any chance of this PR breaking any script for  jenkins-ci-tdx-x86_64-cc_cri_containerd_k8s-qemu ?

**Comment by wainersm** [comment]
@kata-containers/intel someone on Intel with access to  jenkins-ci-tdx-x86_64-cc_cri_containerd_k8s-qemu  could please check if this PR breaks anything on CI?

**Comment by Alex-Carter01** [comment]
> @Alex-Carter01 any chance of this PR breaking any script for jenkins-ci-tdx-x86_64-cc_cri_containerd_k8s-qemu ?

Personally i dont know why this would break the tdx ci, but i am not to knowledgeable of how it works.

**Review by wainersm — APPROVED** [review]
